src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -1116,4 +1101,39 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     Set(welderImageOpt, Some(jupyterImage), rstudioImageOpt).flatten
   }
 
+  private def shouldDeployWelder(cluster: Cluster): Boolean = {
+    !cluster.welderEnabled && dataprocConfig.deployWelderLabel.exists(cluster.labels.contains)
+  }
+
+  private def enableWelder(cluster: Cluster): IO[Cluster] = {
+    for {
+      _ <- IO(logger.info(s"Deploying welder on cluster ${cluster.projectNameString}"))
```

Comment:
shall we log cluster name as well

Reply:
`projectNameString` logs both project name and cluster name.